Option to send Equinox Framework trace to the OSGi LogService

tjwatson opened the pull request:
The trace messages in the framework are currently sent to the System.out.  This commit adds an option to send the Framework trace messages to the OSGi LogService using the LogLevel of Trace.  The Logger name used is the trace option key.

To set Equinox trace options with LoggerAdmin the logger name EQUINOX.TRACE is set to LogLevel TRACE. When EQUINOX.TRACE logger is set to TRACE then all other logger names in the LoggerContext configuration set to TRACE are also assumed to be trace names to be enabled.

tjwatson commented:
@laeubi did you intend to force-push some other changes to my PR.  The changes you made are unrelated to my PR:

https://github.com/eclipse-equinox/equinox/compare/d4af2ac6ed1c831d2c1b55101d58efab87a02e54..8f4fe61ba77802638b3d48efb1839f603d854655

laeubi commented:
@tjwatson I just rebased the branch (as it failed before due to the infrastructure outage), if you like you can safely overwrite the changes or rebase yourself (again) on current master!

tjwatson commented:
> @tjwatson I just rebased the branch (as it failed before due to the infrastructure outage), if you like you can safely overwrite the changes or rebase yourself (again) on current master!

Thanks, I figured that out after asking.  Now the builds has a failure in the testcase I introduced, so I need to investigate that.
